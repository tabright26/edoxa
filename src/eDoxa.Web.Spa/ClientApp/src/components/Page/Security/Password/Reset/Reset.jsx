import React from "react";
import { Card, CardBody, Col, Container, Row } from "reactstrap";
import { Redirect } from "react-router-dom";
import SecurityPassword from "../../../../../forms/Security/Password";
import withUserContainer from "../../../../../containers/App/User/withUserContainer";
import queryString from "query-string";

const ResetPassword = ({ location, actions }) => {
  const code = queryString.parse(location.search).code;
  if (!code) {
    return <Redirect to="/errors/404" />;
  }
  return (
    <div className="app flex-row align-items-center">
      <Container>
        <Row className="justify-content-center">
          <Col md="9" lg="7" xl="6">
            <Card className="mx-4">
              <CardBody className="p-4">
                <h1>Reset Password</h1>
                <p className="text-muted">Reset your account password</p>
                <SecurityPassword.Reset onSubmit={fields => actions.resetPassword(fields, code)} />
              </CardBody>
            </Card>
          </Col>
        </Row>
      </Container>
    </div>
  );
};

export default withUserContainer(ResetPassword);
