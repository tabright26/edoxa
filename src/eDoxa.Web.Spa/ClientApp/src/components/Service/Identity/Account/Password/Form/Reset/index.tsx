import React, { FunctionComponent } from "react";
import {
  FormGroup,
  Form,
  InputGroup,
  InputGroupAddon,
  InputGroupText
} from "reactstrap";
import { Field, reduxForm, InjectedFormProps, FormErrors } from "redux-form";
import Button from "components/Shared/Button";
import Input from "components/Shared/Input";
import { RESET_USER_PASSWORD_FORM } from "utils/form/constants";
import { compose } from "recompose";
import { ValidationSummary } from "components/Shared/ValidationSummary";
import { throwSubmissionError } from "utils/form/types";
import { resetUserPassword } from "store/actions/identity";
import { PASSWORD_REQUIRED } from "utils/form/validators";
import { AxiosActionCreatorMeta } from "utils/axios/types";
import { MapStateToProps, connect } from "react-redux";
import { RootState } from "store/types";
import queryString, { ParseOptions } from "query-string";
import { withRouter, RouteComponentProps } from "react-router-dom";
import { push } from "connected-react-router";
import { UserId } from "types/identity";

type FormData = {
  code: string;
  userId: UserId;
  password: string;
  newPassword: string;
};

type Params = {
  userId: UserId;
};

type OwnProps = RouteComponentProps<Params>;

type StateProps = {};

type OutterProps = {};

type InnerProps = InjectedFormProps<FormData, Props> & StateProps;

type Props = InnerProps & OutterProps;

const Reset: FunctionComponent<Props> = ({
  handleSubmit,
  error,
  submitting,
  anyTouched
}) => (
  <Form onSubmit={handleSubmit}>
    <ValidationSummary anyTouched={anyTouched} error={error} />
    <Field type="hidden" name="code" component={Input.Text} />
    <Field type="hidden" name="userId" component={Input.Text} />
    <InputGroup size="sm" className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="password"
        placeholder="New password"
        autoComplete="password"
        component={Input.Password}
      />
    </InputGroup>
    <InputGroup size="sm" className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="newPassword"
        placeholder="Confirm new password"
        autoComplete="new-password"
        component={Input.Password}
      />
    </InputGroup>
    <FormGroup className="mb-0">
      <Button.Submit loading={submitting} block>
        Reset
      </Button.Submit>
    </FormGroup>
  </Form>
);

const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (
  _state,
  ownProps
) => {
  const options: ParseOptions = {
    decode: false
  };
  const { code } = queryString.parse(ownProps.location.search, options);
  return {
    initialValues: { code, userId: ownProps.match.params.userId }
  };
};

const enhance = compose<InnerProps, OutterProps>(
  withRouter,
  connect(mapStateToProps),
  reduxForm<FormData, Props>({
    form: RESET_USER_PASSWORD_FORM,
    onSubmit: async (values, dispatch) => {
      try {
        return await new Promise((resolve, reject) => {
          const meta: AxiosActionCreatorMeta = { resolve, reject };
          dispatch(resetUserPassword(values, meta));
        });
      } catch (error) {
        throwSubmissionError(error);
      }
    },
    onSubmitSuccess: (_result, dispatch) => {
      dispatch(push("/authentication/login"));
    },
    validate: values => {
      const errors: FormErrors<FormData> = {};
      if (!values.password) {
        errors.password = PASSWORD_REQUIRED;
      } else if (values.password !== values.newPassword) {
        errors._error = "Confirm password doesn't match password";
      }
      return errors;
    }
  })
);

export default enhance(Reset);
