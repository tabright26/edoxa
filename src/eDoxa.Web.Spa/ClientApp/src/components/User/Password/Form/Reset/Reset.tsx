import React, { FunctionComponent } from "react";
import {
  FormGroup,
  Form,
  InputGroup,
  InputGroupAddon,
  InputGroupText
} from "reactstrap";
import { Field, reduxForm, InjectedFormProps } from "redux-form";
import Button from "components/Shared/Button";
import Input from "components/Shared/Input";
import { RESET_USER_PASSWORD_FORM } from "forms";
import { compose } from "recompose";
import FormValidation from "components/Shared/Form/Validation";
import { throwSubmissionError } from "utils/form/types";
import { resetUserPassword } from "store/actions/identity/actions";
import {
  emailRegex,
  passwordRegex,
  EMAIL_REQUIRED,
  EMAIL_INVALID,
  PASSWORD_REQUIRED,
  PASSWORD_INVALID
} from "validation";
import { AxiosActionCreatorMeta } from "utils/axios/types";

interface Props {}

interface FormData {}

const validate = values => {
  const errors: any = {};
  if (!values.email) {
    errors.email = EMAIL_REQUIRED;
  } else if (!emailRegex.test(values.email)) {
    errors.email = EMAIL_INVALID;
  }
  if (!values.password) {
    errors.password = PASSWORD_REQUIRED;
  } else if (!passwordRegex.test(values.password)) {
    errors.password = PASSWORD_INVALID;
  }
  return errors;
};

async function submit(values, dispatch) {
  try {
    return await new Promise((resolve, reject) => {
      const meta: AxiosActionCreatorMeta = { resolve, reject };
      dispatch(resetUserPassword(values, meta));
    });
  } catch (error) {
    throwSubmissionError(error);
  }
}

const ResetUserPasswordForm: FunctionComponent<InjectedFormProps<FormData> &
  Props &
  any> = ({ handleSubmit, handleCancel, dispatch, error }) => (
  <Form
    onSubmit={handleSubmit(data =>
      submit(data, dispatch).then(() => handleCancel())
    )}
  >
    {error && <FormValidation error={error} />}
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>@</InputGroupText>
      </InputGroupAddon>
      <Field type="text" name="email" label="Email" component={Input.Text} />
    </InputGroup>
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="password"
        label="Password"
        component={Input.Password}
      />
    </InputGroup>
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field
        type="password"
        name="confirmPassword"
        label="Confirm Password"
        component={Input.Password}
      />
    </InputGroup>
    <FormGroup className="mb-0">
      <Button.Submit block>Reset</Button.Submit>
    </FormGroup>
  </Form>
);

const enhance = compose<any, any>(
  reduxForm<FormData, Props>({
    form: RESET_USER_PASSWORD_FORM,
    validate
  })
);

export default enhance(ResetUserPasswordForm);
