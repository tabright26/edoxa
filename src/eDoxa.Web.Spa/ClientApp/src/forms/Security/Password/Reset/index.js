import ResetPasswordForm from "./Reset";

export default ResetPasswordForm;
