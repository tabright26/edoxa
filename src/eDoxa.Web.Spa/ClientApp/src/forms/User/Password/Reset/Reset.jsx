import React from "react";
import { Field, reduxForm } from "redux-form";
import { FormGroup, Form, InputGroup, InputGroupAddon, InputGroupText } from "reactstrap";
import validate from "./validate";
import Button from "../../../../components/Shared/Override/Button";
import Input from "../../../../components/Shared/Override/Input";
import { RESET_PASSWORD_FORM } from "../../../../forms";

const ResetPasswordForm = ({ handleSubmit }) => (
  <Form onSubmit={handleSubmit}>
    <Field
      type="text"
      name="email"
      label="Email"
      component={props => (
        <InputGroup className="mb-3">
          <InputGroupAddon addonType="prepend">
            <InputGroupText>@</InputGroupText>
          </InputGroupAddon>
          <Input.Text {...props} />
        </InputGroup>
      )}
    />
    <Field
      type="password"
      name="password"
      label="Password"
      component={props => (
        <InputGroup className="mb-3">
          <InputGroupAddon addonType="prepend">
            <InputGroupText>
              <i className="icon-lock"></i>
            </InputGroupText>
          </InputGroupAddon>
          <Input.Password {...props} />
        </InputGroup>
      )}
    />
    <Field
      type="password"
      name="confirmPassword"
      label="Confirm Password"
      component={props => (
        <InputGroup className="mb-3">
          <InputGroupAddon addonType="prepend">
            <InputGroupText>
              <i className="icon-lock"></i>
            </InputGroupText>
          </InputGroupAddon>
          <Input.Password {...props} />
        </InputGroup>
      )}
    />
    <FormGroup className="mb-0">
      <Button.Submit block>Reset</Button.Submit>
    </FormGroup>
  </Form>
);

export default reduxForm({ form: RESET_PASSWORD_FORM, validate })(ResetPasswordForm);
