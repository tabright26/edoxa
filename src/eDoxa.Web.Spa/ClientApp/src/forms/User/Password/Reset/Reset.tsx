import React, { FunctionComponent } from "react";
import { FormGroup, Form, InputGroup, InputGroupAddon, InputGroupText } from "reactstrap";
import { Field, reduxForm } from "redux-form";
import Button from "components/Shared/Override/Button";
import Input from "components/Shared/Override/Input";
import { RESET_USER_PASSWORD_FORM } from "forms";
import { validate } from "./validate";
import { compose } from "recompose";

const ResetUserPasswordForm: FunctionComponent<any> = ({ handleSubmit, resetUserPassword }) => (
  <Form onSubmit={handleSubmit(data => resetUserPassword(data))}>
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>@</InputGroupText>
      </InputGroupAddon>
      <Field type="text" name="email" label="Email" component={Input.Text} />
    </InputGroup>
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field type="password" name="password" label="Password" component={Input.Password} />
    </InputGroup>
    <InputGroup className="mb-3">
      <InputGroupAddon addonType="prepend">
        <InputGroupText>
          <i className="icon-lock"></i>
        </InputGroupText>
      </InputGroupAddon>
      <Field type="password" name="confirmPassword" label="Confirm Password" component={Input.Password} />
    </InputGroup>
    <FormGroup className="mb-0">
      <Button.Submit block>Reset</Button.Submit>
    </FormGroup>
  </Form>
);

const enhance = compose<any, any>(reduxForm<any, { handleCancel: () => any }, string>({ form: RESET_USER_PASSWORD_FORM, validate }));

export default enhance(ResetUserPasswordForm);
