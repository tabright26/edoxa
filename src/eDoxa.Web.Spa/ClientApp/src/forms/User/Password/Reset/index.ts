import ResetUserPasswordContainer from "./Container";

export default ResetUserPasswordContainer;
