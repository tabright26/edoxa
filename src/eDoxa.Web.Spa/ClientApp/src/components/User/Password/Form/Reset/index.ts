import Form from "./Reset";

export default Form;
