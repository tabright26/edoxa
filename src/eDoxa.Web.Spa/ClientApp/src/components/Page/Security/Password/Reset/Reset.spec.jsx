import React from "react";
import renderer from "react-test-renderer";
import Reset from "./Reset";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Provider
        store={{
          getState: () => {
            return {
              oidc: {
                user: null
              }
            };
          },
          dispatch: action => {},
          subscribe: () => {}
        }}
      >
        <MemoryRouter>
          <Reset location={{ search: "?code=test" }} />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
