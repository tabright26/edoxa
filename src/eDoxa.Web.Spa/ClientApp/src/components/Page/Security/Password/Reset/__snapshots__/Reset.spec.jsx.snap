// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="app flex-row align-items-center"
>
  <div
    className="container"
  >
    <div
      className="justify-content-center row"
    >
      <div
        className="col-md-9 col-lg-7 col-xl-6"
      >
        <div
          className="mx-4 card"
        >
          <div
            className="p-4 card-body"
          >
            <form
              className=""
            >
              <h1>
                Reset Password
              </h1>
              <p
                className="text-muted"
              >
                Reset your account password
              </p>
              <div
                className="mb-3 input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                  >
                    @
                  </span>
                </div>
                <input
                  autoComplete="email"
                  className="form-control"
                  placeholder="Email"
                  type="text"
                />
              </div>
              <div
                className="mb-3 input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                  >
                    <i
                      className="icon-lock"
                    />
                  </span>
                </div>
                <input
                  autoComplete="new-password"
                  className="form-control"
                  placeholder="Password"
                  type="password"
                />
              </div>
              <div
                className="mb-4 input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                  >
                    <i
                      className="icon-lock"
                    />
                  </span>
                </div>
                <input
                  autoComplete="new-password"
                  className="form-control"
                  placeholder="Repeat password"
                  type="password"
                />
              </div>
              <button
                aria-label={null}
                className="btn btn-primary btn-block"
                onClick={[Function]}
              >
                Reset Password
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
